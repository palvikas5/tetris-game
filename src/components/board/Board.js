import React from "react";
import { BOARD_SIZE } from "../../utils/constants";
import BrickController from "./../brick-v2/BrickController";
import BoardRow from "./BoardRow";

const Board = ({ canPlayGame }) => {
  return (
    <div className="board">
      {[...Array(BOARD_SIZE.row)].map((e, index) => {
        return <BoardRow key={index} row={index} />;
      })}
      <BrickController canPlayGame={canPlayGame} />
    </div>
  );
};

export default Board;
