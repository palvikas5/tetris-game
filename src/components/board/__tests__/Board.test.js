import React from "react";
import { render } from "@testing-library/react";
import Board from "../Board";

describe("<Board />", () => {
  it("should render successfully", () => {
    const { container } = render(<Board />);
    expect(container).toMatchSnapshot();
  });
});
