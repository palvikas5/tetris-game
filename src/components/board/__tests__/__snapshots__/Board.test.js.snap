// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> should render successfully 1`] = `
<div>
  <div
    class="board"
  >
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="board-cell"
    />
    <div
      class="bricks-container"
    />
  </div>
</div>
`;
